src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UnaryOperator.cs

```diff
@@ -485,6 +485,7 @@ private static bool IsIndirectOrInstanceField(BoundExpression expression)
                     return ((BoundLocal)expression).LocalSymbol.RefKind != RefKind.None;
 
                 case BoundKind.Parameter:
+                    Debug.Assert(!IsCapturedPrimaryConstructorParameter(expression));
```

Comment:
>Debug.Assert(!IsCapturedPrimaryConstructorParameter(expression)); [](http://example.com/codeflow?start=20&length=65)

How do we know the expression is not a primary constructor parameter?

Reply:
> How do we know the expression is not a primary constructor parameter?

The ```expression``` is supposed to be lowered before we get here. Captured parameters are replaced with fields during lowering.